hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/scale/AbstractSTestS3AHugeFiles.java

```diff
@@ -119,6 +119,14 @@ protected Configuration createScaleConfiguration() {
         DEFAULT_HUGE_PARTITION_SIZE);
     assertTrue("Partition size too small: " + partitionSize,
         partitionSize >= MULTIPART_MIN_SIZE);
+    removeBaseAndBucketOverrides(conf,
+        SOCKET_SEND_BUFFER,
+        SOCKET_RECV_BUFFER,
+        MIN_MULTIPART_THRESHOLD,
+        MULTIPART_SIZE,
+        USER_AGENT_PREFIX,
+        FAST_UPLOAD_BUFFER);
```

Comment:
random comment on maintainability of the tests in S3A

would a wrapper for `Configuration` make more sense, where this wrapper would remove base/bucket overrides for any method you set?

This wrapper would just intercept `Configuration::set(String name, String value, String source)`, with a body like:

```java
/** Wrapper for Configuration which drops any bucket overrides when setting a configuration **/
public TestConfiguration extends Configuration {
    private Configuration innerConfiguration;

    public TestConfiguration(Configuration conf) {
        this.innerConfiguration = conf;
    }

    public void set(String name, String value, String source) {
        this.removeBaseAndBucketOverrides(name);
        this.innerConfiguration.set(name, value, source);
    }
}
```

would be a mess to update all existing tests of course

Reply:
oh that's cute. Be hard to get right though, especially as there are other troublespots related to default option loading (if anyone creates an HdfsConfiguration it forces hdfs-default.xml onto the classpath which reloads things and breaks tests)